profiling/simple-kernel-timer/kp_kernel_timer.cpp

```diff
@@ -52,23 +57,90 @@ void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
 void kokkosp_finalize_library() {
   double finishTime = seconds();
 
+  const char* kokkos_tools_timer_json_raw = getenv("KOKKOS_TOOLS_TIMER_JSON");
+  const bool kokkos_tools_timer_json =
+      kokkos_tools_timer_json_raw == NULL
+          ? false
+          : strcmp(kokkos_tools_timer_json_raw, "1") == 0 ||
+                strcmp(kokkos_tools_timer_json_raw, "true") == 0 ||
+                strcmp(kokkos_tools_timer_json_raw, "True") == 0;
+
+  double kernelTimes = 0;
+
   char* hostname = (char*)malloc(sizeof(char) * 256);
   gethostname(hostname, 256);
 
   char* fileOutput = (char*)malloc(sizeof(char) * 256);
-  snprintf(fileOutput, 256, "%s-%d.dat", hostname, (int)getpid());
+  snprintf(fileOutput, 256, "%s-%d.%s", hostname, (int)getpid(),
+           kokkos_tools_timer_json ? "json" : "dat");
 
   free(hostname);
   FILE* output_data = fopen(fileOutput, "wb");
 
+  std::vector<KernelPerformanceInfo*> kernelList;
+
   const double totalExecuteTime = (finishTime - initTime);
-  fwrite(&totalExecuteTime, sizeof(totalExecuteTime), 1, output_data);
+  if (!kokkos_tools_timer_json) {
+    fwrite(&totalExecuteTime, sizeof(totalExecuteTime), 1, output_data);
 
-  std::vector<KernelPerformanceInfo*> kernelList;
+    for (auto kernel_itr = count_map.begin(); kernel_itr != count_map.end();
+         kernel_itr++) {
+      kernel_itr->second->writeToBinaryFile(output_data);
+    }
+  } else {
+    for (auto kernel_itr = count_map.begin(); kernel_itr != count_map.end();
+         kernel_itr++) {
+      kernelList.push_back(kernel_itr->second);
+      kernelTimes += kernel_itr->second->getTime();
+    }
+
+    std::sort(kernelList.begin(), kernelList.end(),
```

Comment:
It seems that sorting the kernels wasn't a feature of `kp_kernel_timer`, but was used in `kp_kernel_timer_json`.

I think sorting or not needs to be clearly stated. What do you think @vlkale?

IMO sorting isn't a requirement and can be done by the user if need be.

Reply:
I mean, for this tool, sorting by (I guess) increasing time, isn't needed.

Reply:
Whether to sort or not is beyond the scope of this PR.

Reply:
`IMO sorting isn't a requirement and can be done by the user if need be.` just to say, If we can avoid post-processing of those .json and get the information easily just by opening them, this is better. I guess it could be an option KOKKOS_TOOLS_TIMER_SORT though

Reply:
IMO if the tool documentation is not saying "results are always sorted by increasing time", it shouldn't be.

Also, maybe you might want to see your regions sorted by increasing time, but I might want to see the regions in their execution order.

So indeed, it seems your `KOKKOS_TOOLS_TIMER_SORT` option would be a good compromise :wink: